### PR TITLE
PALS-1534: Update the Artstor AJI Modal/Banner with copy, and a list of users to roll out to

### PR DESCRIPTION
**Pull request by ssharma2ithaka**
Resolves PALS-1534

- Update logic for displaying AJI intercept and post login banner to targeted users
- Add data-sc attributes for GA tracking

AJI Intercept
image

Post-login drop down banner
image

Link behavior on hover
image

### CODE REVIEW

File: src/app/_services/auth.service.ts [hunk 1]
```diff
@@ -421,10 +421,11 @@ export class AuthService implements CanActivate {
    public showAJIIntercept(): boolean {
     let user = this.getUser()
     let institution = this.institutionObjSource.value
+    let showIntercept = user.preference.showAJIModalAndBanner ? user.preference.hasOwnProperty('showAJIModalAndBanner') : false
     if (this.getFromStorage('AJIInterceptClosed')) {
       return false
     }
-    return user && user.isLoggedIn && institution.institutionName && institution.institutionName == 'JSTOR'
+    return user && user.isLoggedIn && showIntercept && institution.institutionName && institution.institutionName == 'JSTOR'
```
Comment (adandavati) [inline src/app/_services/auth.service.ts:428]:
```suggestion
    return user && user.isLoggedIn && user.preference.hasOwnProperty('showAJIModalAndBanner') && user.preference.showAJIModalAndBanner
```

File: src/app/_services/auth.service.ts [hunk 2]
```diff
@@ -421,10 +421,11 @@ export class AuthService implements CanActivate {
    public showAJIIntercept(): boolean {
     let user = this.getUser()
     let institution = this.institutionObjSource.value
+    let showIntercept = user.preference.showAJIModalAndBanner ? user.preference.hasOwnProperty('showAJIModalAndBanner') : false
```
Comment (adandavati) [inline src/app/_services/auth.service.ts]:
```suggestion
```

File: src/app/_services/auth.service.ts [hunk 3]
```diff
@@ -433,11 +434,12 @@ export class AuthService implements CanActivate {
    */
   public showPostLoginBanner(): boolean {
     let user = this.getUser()
+    let showBanner = user.preference.showAJIModalAndBanner ? user.preference.hasOwnProperty('showAJIModalAndBanner') : false
```
Comment (adandavati) [inline src/app/_services/auth.service.ts]: I don't think we need to check this again here.

File: src/app/_services/auth.service.ts [hunk 4]
```diff
@@ -421,10 +421,11 @@ export class AuthService implements CanActivate {
    public showAJIIntercept(): boolean {
     let user = this.getUser()
     let institution = this.institutionObjSource.value
+    let showIntercept = user.preferences.showAJIModalAndBanner ? user.preferences.hasOwnProperty('showAJIModalAndBanner') : false
```
Comment (adandavati) [inline src/app/_services/auth.service.ts:424]:
```suggestion

```

File: src/app/modals/aji-intercept/aji.component.ts [hunk 5]
```diff
@@ -31,16 +32,38 @@ export class AJIInterceptModal implements OnInit {
...
+        }
+      )).subscribe()
+  }
+
   public dismissModal(): void {
     this._auth.store('AJIInterceptClosed', true);
     this.closeModal.emit()
   }
 
+  public remindMeLater(): void {
```
Comment (adandavati) [inline src/app/modals/aji-intercept/aji.component.ts:57]: If the user clicks remind me later we don't want to do the following. This should only happen if they dimiss the modal by clicking X.

Comment (adandavati) [reply src/app/modals/aji-intercept/aji.component.ts:57]: If they click remind me later we just want to close the modal for this session.

File: src/app/_services/auth.service.ts (hunk 3 above)
Comment (ssharma2ithaka) [reply src/app/_services/auth.service.ts]: Not sure if I follow correctly. Do we want to let go of the check entirely?

**Pull request merged**